healthCheck.ps1

```diff
@@ -0,0 +1,127 @@
+    $az_installed = $true
+    break}
+     }
+
+if ($az_installed -eq $false)
+{
+write-host "Azure CLI is not installed - script can not be executed"
+Exit 
+}
+
+write-host "Check if iot-extension for AZ CLI installed"
+$az_iot_ext_install_status=$(az extension show --name azure-cli-iot-ext)
+if ($az_iot_ext_install_status.Length -eq 0)
+    {
+        write-host "Add iot-extension for AZ CLI"
+        az extension add --name azure-cli-iot-ext
+    }
+
+# Sender wait duration
+$senderWaitValue = 2000
+
+# Listener wait duration
+$listenerWaitValue = 2000
+
+
+#Setup the listener process info
+$startInfo = New-Object System.Diagnostics.ProcessStartInfo
+$startInfo.FileName = "cmd.exe"
+
+#get approximate invocation time
+$epochseconds = Get-Date (Get-Date).ToUniversalTime() -UFormat %s
+$invokeTimeInEpochMilliSeconds= [double][math]::Round([double]$epochseconds*1000)
+
+$startInfo.Arguments = "/c az iot hub monitor-events --props app -y -n "+$iotHubName+" -d "+ $deviceId+" -l "+$iotHubCS + " -e "+$invokeTimeInEpochMilliSeconds
+$startInfo.RedirectStandardOutput = $true
+
+$startInfo.RedirectStandardError = $true
+$startInfo.UseShellExecute = $false
+$startInfo.CreateNoWindow = $true
+
+
+#Start the listener process
+$process = New-Object System.Diagnostics.Process
+$process.StartInfo = $startInfo
+$process.Start() | Out-Null
+
+#Setup the sender process info
+$senderStartInfo = New-Object System.Diagnostics.ProcessStartInfo
+$senderStartInfo.FileName = "cmd.exe"
+
+#create a correlation ID for the sender
+$random = get-random -inputobject (1..40000) 
+$correlationid = "CId"+[string]$random
+
+$methodpayload = "{\" +"`"correlationId\`":\`""+$correlationid+"\`"}"
+$senderStartInfo.Arguments = "/c az iot hub invoke-module-method  -n "+$iotHubName+" -d "+$deviceId+" -m "+$moduleId+" --mn "+$methodName+" -l "+$iotHubCS +" --mp "+ $methodpayload
+
+$senderStartInfo.RedirectStandardOutput = $true
+$senderStartInfo.UseShellExecute = $false
+$senderStartInfo.CreateNoWindow = $true
+
+#Start the sender process
+$senderProcess = New-Object System.Diagnostics.Process
+$senderProcess.StartInfo = $senderStartInfo
+
+Write-Host "Test commnunication from IOT Hub:$iotHubName to module: $moduleId on device:$deviceId by invoke module direct method:$methodName ..."
+
+$senderProcess.Start() | Out-Null
+$senderOutput = $senderProcess.StandardOutput.ReadToEnd()
+$senderProcess.WaitForExit($senderWaitValue)
+
+
+ if($senderOutput.Contains("200") -and $senderOutput.Contains($correlationid)) 
+ {
+    Write-Host "Module direct Method Invoked - Hub to Device connection validated" -ForegroundColor Green
+    write-host $senderOutput
+ }
+ else
+ {
+    Write-Host "Module direct Method Invoked - Hub to Device connection validation failed:"+$senderOutput -ForegroundColor Red
+    exit 1
+
+ }
+
+$senderProcess.close()
+$senderProcess.Dispose()
+
+$listenerOutput = $process.StandardOutput.ReadToEnd()
+
+$process.WaitForExit($listenerWaitValue)

```

Comment:
Is the 2000ms timeout enough for the message to arrive? The major consideration here is: if something caused high latency for the messages, we don't expect the health check to fail. Maybe we could try several times before fail the health check. Technically the first release pipeline execution after sample project created should always success.

Reply:
I added a param to configure the wait time for the listener to receive messages from the IOThub. the  that will ensure the listener waits the max needed time to verify before failing the test if the message was not received. that should cover that issue